tough/src/editor/mod.rs

```diff
@@ -394,4 +621,127 @@ impl RepositoryEditor {
             _extra: HashMap::new(),
         }
     }
+
+    /// Refreshes the timestamp and snapshot
+    /// Allows for expirations to be changed
+    pub fn update_snapshot<T>(
```

Comment:
Perhaps I haven't read far enough yet, but why would we need this? The reason we discard the expiration and version when reading a repository is so that both get updated. It seems kinda odd to offer a method to update snapshot and timestamp together as we offer "setter" methods for both.